docs: add design pages for runtime recording feature (#118)
tier4/CARET_doc: docs/design/trace_points/initialization_trace_points.md

```

### Tracepoints for representing structure of executor and callback group

> A handler such as `timer_handle` and `subscription_handle` are assigned to a callback group. A callback group belongs to an executor.

Reviewer: ```suggestion
A handler such as `timer_handle` or `subscription_handle` is assigned to a callback group. A callback group belongs to an executor.
```

Reviewer: fixed at https://github.com/hsgwa/CARET_doc/pull/6/files
